alchemist/alchemist-incarnation-biochemistry/src/main/java/it/unibo/alchemist/model/implementations/actions/RemoveJunctionInCell.java

```diff
@@ -46,7 +43,7 @@
      * @param e the environment
      * @param rg the random generator
      */
-    public RemoveJunctionInCell(final Junction junction, final ICellNode n, final Environment<Double> e, final RandomGenerator rg) {
+    public RemoveJunctionInCell(final Environment<Double> e, final CellNode n, final Junction junction, final RandomGenerator rg) {
```

Comment:
I suggest (as you did in other classes) to accept Node<Double>, then internally check with instanceof, and throw an exception if something's wrong.